src/services/widgets/widgets.svc.coffee

```diff
@@ -74,22 +74,39 @@ angular
         setting.initialize()
       return true
 
-    @updateWidgetSettings = (widget, needContentReload=true) ->
+    @updateWidgetSettings = (widget, needContentReload=true, ignoreReach=false) ->
       widget.isEditMode = false
-
       if _.isEmpty(widget.settings)
         $log.warn("ImpacWidgetsSvc: Tried to update widget: #{widget.id} with no settings", widget)
         return false
 
+      changedGlobalSetting = _.find widget.settings, (setting)-> setting.reach == 'dashboard'
+
+      if (changedGlobalSetting && !ignoreReach)
+        return _self.updateAllSameWidgets(widget, ImpacDashboardsSvc.getCurrentDashboard(), changedGlobalSetting)
+
       widget.isLoading = true if needContentReload
       meta = _.reduce(_.map(widget.settings, (set) -> set.toMetadata() ), (result, setMeta) -> angular.merge(result, setMeta))
-      
-      return _self.update(widget, { metadata: meta }).then(
+
+      _self.update(widget, { metadata: meta }).then(
         (updatedWidget) ->
           if needContentReload
             _self.show(updatedWidget).finally( -> updatedWidget.isLoading = false )
       )
 
+    @updateAllSameWidgets = (widget, dashboard, settings) ->
```

Comment:
1) I can't do that, because 2 services, according to angular, can't be in dependency of each other in the same time, so, I can't use WidgetsSvc methods from DashboardSvc directly. 
All I can is set method updateAllSameWidgets as a callback, so it will be called from DashbordSvc after dashboard updated with updateSetting method.
2) So...then widgets required them own key parameter. Because if I rewrite metadata by setting key I rewrite it in all widgets where the setting with same key is defined, even if the widget uses another list of parameters. In other words, I will rewrite settings of "sales-cycle" when saves "sales-opportunities-funnel", for example.